### ISSUE TITLE
Wrong token on password reset

### ISSUE THREAD

**Issue by kesanf** [AzureAD/microsoft-authentication-library-for-objc#833]
When password reset flow if finished object returned is MSALResult which contains accessToken, expiresOn, etc. This access token is not valid, so I have to login again instead just continue using this token. Is there a way to get valid access token without additional requests?

**Comment by rohitnarula7176** [comment]
@kesanf Thank you for reporting this. In B2C the access tokens returned are policy specific so if you get an access token from the password reset endpoint, you cannot use that access token to access any other resource. Can you provide more step-by-step details on your flow so that we can investigate further.

**Comment by kesanf** [comment]
Is there a way after password reset to be logged in?

**Comment by rohitnarula7176** [comment]
@kesanf Please let me know if the scenario you are describing is as follows:

1. Call acquireTokenInteractive by passing authority "sign_in" which results in an Access Token, Refresh Token and Account (whose object Id suffix matches the policy ("sign_in").

2. Call acquireTokenInteractive passing authority "password_reset" which results in an Access Token, Refresh Token and Account (whose object Id suffix matches the policy ("password_reset")

3. Call acquireTokenSilent passing authority "sign_in", resource and account that matches sign_in policy from step 1, Msal will find and return a valid Access Token which stays valid for an hour and you can use this token to access the resource.

4. But after one hour when Access Token expires, if you call acquireTokenSilent by passing same authority you passed in Step 3, MSAL will no longer return Access token as it is expired and will try to use the Refresh Token (received in step 1 before you reset your password) to refresh the access token and will fail because of the error "The provided grant has been revoked. Please reauthenticate and try again." This is expected behavior as the RT we are trying to use gets revoked after you reset your password and you can no longer use it.

In summary, you can stay signed in after password reset as long as the access token is valid (which is 1 hour in most cases and this time is configurable) but after this access token expires, you will need to call acquireTokenInteractive again by passing sign_in policy to retrieve a new list of tokens for your app.

**Comment by kesanf** [comment]
This scenario:

1. Call acquireTokenInteractive by passing authority "sign_in" which results in an Access Token, Refresh Token and Account (whose object Id suffix matches the policy ("sign_in").

2. Call acquireTokenInteractive passing authority "password_reset" which results in an Access Token, Refresh Token and Account (whose object Id suffix matches the policy ("password_reset")

3. Now when I have new password I have to use step 1. again, right? Token that I had from step 1. should be invalidated on your side once my password is reseted?

**Comment by oldalton** [comment]
@kesanf. Yes, step nr 3 requires you to use the policy from step nr 1. Regarding token invalidation, @nickgmicrosoft can provide more clarity on this.

**Comment by kesanf** [comment]
Thank you very much. You can close the issue.
